pact/pact.py

```diff
@@ -103,26 +100,20 @@ def given(self, provider_state):
         :type provider_state: basestring
         :rtype: Pact
         """
-        self._provider_state = provider_state
+        self._interactions.insert(0, {'provider_state': provider_state})
```

Comment:
I'm fine with the code as is, but my curiosity streak wants to ask: Was there a particular case that stood out that led to inserting the new interactions at the beginning of the list instead of using `append` and `self._interactions[-1]`?